src/diagram/components/diagram.tsx

```diff
@@ -85,26 +85,49 @@ export const _Diagram = ({ dqRoot, getDiagramExport, hideControls, hideNavigator
     }
   };
 
+  const deleteEdge = (edge: Edge) => {
+    const { source, target } = edge;
+    const targetModel = dqRoot.getNodeFromVariableId(target);
+    const sourceModel = dqRoot.getNodeFromVariableId(source);
+    sourceModel?.tryVariable && targetModel?.removeInput(sourceModel.variable);
+  };
+
+  const deleteAllNodesEdges = (variableId: string) => {
```

Comment:
Function name reads a bit awkwardly to me -- perhaps `deleteAllEdgesOfNode()`?

Reply:
Yeah, `deleteAllEdgesofNode()` is better. Thanks!